### DISCUSSION TITLE
Alarm Conditioned by an Event Button

### DISCUSSION THREAD

**Discussion by DennieSch** [artisan-roaster-scope/artisan#1528] [Q&A] [answered]
Greetings,

I'm wondering if there is straight forward way to condition an Alarm on an Event Button. 
For example, I've created some of my own 'Event Buttons' via the 'Config/Events/Buttons' tab. I'd like to 'condition' the 'activation' of some of my Alarms based on... my Event Buttons. 

After going through the documentation, the only way I see to do this is to have multiple alarm sets (unless I've missed something). Then, use the "alarmset(<as>)" command as part of the Event Button action to swap in/out different alarm sets. I can see this a way but wondering if there might be a way that doesn't involve creating multiple alarm sets. 

Thoughts?

Cheers,
Dennie

**Comment by roasterdave** [comment, marked as answer]
There is not a way to directly connect a button to an alarm.  However, there is a method that will work if you have an unused special event type.  Multiple buttons can trigger separate alarms by setting the event to different values.  In the example below the event is named "AlarmTrig".  

Create one or more buttons to set the event to a specific value.  In the example there are two buttons, one that sets "AlarmTrig" to 10 and the other sets it to 20.  

Create a Virtual Device that is tied to the event's value.  In the example the Virtual Device is named "AlarmTrigger".  "E3" is entered as the symbolic equation for the virtual device since "AlarmTrig".  "E3" i the symbol that represents the 3rd event type.  

Use the Extra Device value as the "Source" to trigger an alarm. That alarm can then be used as the "If alarm" condition for any other alarm(s).  In this example the values 10 and 20 trigger alarms number 1 and number 2 respectively.  However alarm number 2 will not trigger until button 1 is pressed, which fires alarm number 1.  Then the next time button 2 is pressed alarm 2 will trigger.  

Be aware, a cascade of alarms can not happen within the same sample period. The conditioning alarm will happen in one sample period and the conditioned alarm will not trigger until the next sample period. Also alarms will fire only once per roast and cannot be triggered multiple times.

![alarm 1 1_ver007](https://github.com/artisan-roaster-scope/artisan/assets/30009312/2619c2aa-4e39-4554-ad89-e7281295bf91)

**Reply by DennieSch** [reply]
Thanks Dave!
This should work great for me. Thank you very much for your help. (This is a cool approach.)
I'm looking to click/select one of a couple buttons at the beginning of the roast. Then, I can trigger one (or more) alarms based on the value that was assigned to "AlarmTrig"... which would be triggered much later in the roast. 
Thanks again Dave!

Cheers,
Dennie
